### PR TITLE
feat(argo-workflows): Disable leader election on single repl controllers

### PR DESCRIPTION
**Pull request by tico24**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

tl;dr - The controller keeps trying to do Leader Election even if it's on its own. This PR turns that off for single-replica controllers. This reduces the load on the controller and k8s api a small amount. Useful for large-scale deployments.

I'm aware that what I've done is very opinionated. I could make it values-controlled if required, but I thought this would confuse users more than help them. I can't think of any situation where you'd want leader election enabled on a single-repl controller.

### CODE REVIEW

**Review by Joibel — APPROVED** [review]
Looks good to me

**Review by yu-croco — APPROVED** [review]
Good catch! 👏

**Review by vladlosev — APPROVED** [review]
LGTM
